docs/_includes/examples/fakes/__tests.js

```diff
@@ -0,0 +1,146 @@
+"use strict";
+
+var proxyquire = require("proxyquire").noCallThru();
+var sinon = require("../../../../lib/sinon");
+var assert = require("@sinonjs/referee").assert;
+var stubs = {
+    "sinon": {
+        fake: sinon.fake,
+        replace: sinon.replace
+    }
+};
+
+describe("`fake` examples", function () {
+    beforeEach(function () {
+        sinon.replace(global.console, "log", sinon.fake());
+    });
+
+    afterEach(function () {
+        sinon.restore();
+    });
+
+    // basic.js
+    describe("basic invocation", function () {
```

Comment:
These `describe` blocks here serve no purpose. If you want to make the test names more descriptive, I would suggest you just add to the test names. Could we simplify this a bit? It would be nice to use the names in the `proxyquire` as well although that's not necessary. This could be simplified into an object:

```js
{ 'basic - test name': testFn } 
// or
{ 'test name': ['basic' /* filename to load with proxyquire? */, testFn] }

// ...
Object.keys(testObject)
  .forEach(label => {
    it(label, testFn);
  })
;
```

Reply:
Another thought, why don't we read the names of the files in the test directory and just load and run them dynamically?Then we could use the file names as the key in the `testObject` above:

```js
{  './basic': testFn }
```

Reply:
Good points. I'll try to re-work this first version to incorporate these